### PR TITLE
Implementing an auto-logout feature for Android(Kotlin)

### LINKED ISSUE #5413: [LANGUAGES]Implementing an auto-logout feature for Android(Kotlin)
### Topic Suggestion
[LANGUAGES]Implementing an auto-logout feature for Android(Kotlin)

### Writing sample(s): 
My latest article is found [here](https://www.section.io/engineering-education/javascript-math-animations/)

#### Proposed title of article
Implementing an auto-logout feature for Android(Kotlin)

#### Proposed article introduction
Handling sensitive data in an app may require some checks to see if the user is still active and decide whether to keep him logged in or not. We will use a very simple approach to do that - using gesture detectors and the Handler class to set timeouts. The same logic is being used [here](https://www.section.io/engineering-education/autologout-js/), but we will do that for Android.

#### Key takeaways 
The user will learn about:
- Using the Handler class to create an auto-logout feature(though the latter will not be that necessary).
- Using Kotlin to implement this.
- Using gesture detectors to implement this.

#### Article quality
We will do that in a very simple manner without invoking too many classes like in other blogs. We will additionally use [gesture detectors](https://developer.android.com/training/gestures/detector) to do the timer resets.

#### References
N/A

### PR DESCRIPTION
**Pull request by vinstex**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5413 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @vinstex 👋 .
My name is Briana and I will be conducting the review for your article.
Just one small thing I want you to correct as I proceed with preliminary revisions. I need you to scale down your images to the required 600 x 400 px. You can use [BeFunky](https://www.befunky.com/) to do this.

**Comment by vinstex** [comment]
Greetings Nzivu. I used to think it's only the hero image that needs to be scaled down. Anyway, I'm doing it right away. My author folder is `vincent-ngunzulu`

**Comment by vinstex** [comment]
Or I can remove the screenshot part. Because they really look awkward when I reduce the to that size. @BrianaNzivu

**Comment by hectorkambow** [comment]
@section-engineering-education/content-moderator-team 
Team lets keep an eye on this PR for quality - as the author has expressed an interest in becoming a peer reviewer. 

I've mentioned the 1st step would be to publish an article with little to no revisions required. 

cc @BrianaNzivu  

🚀  

Thank you

**Comment by BrianaNzivu** [comment]
> @section-engineering-education/content-moderator-team 
> Team lets keep an eye on this PR for quality - as the author has expressed an interest in becoming a peer reviewer. 
> 
> I've mentioned the 1st step would be to publish an article with little to no revisions required. 
> 
> cc @BrianaNzivu  
> 
> 🚀  
> 
> Thank you 

Noted 👍🏾

**Comment by vinstex** [comment]
@BrianaNzivu . there is some error I did during the typing of the document. I put a return value on some two methods which didn't any return type😃. Noticed? I will change that when you are done with some revisions😃😃

**Comment by lalith1403** [comment]
plagiarism check done ✔️

**Comment by BrianaNzivu** [comment]
> @BrianaNzivu . there is some error I did during the typing of the document. I put a return value on some two methods(`onLongPress()` and `onShowPress()`) which didn't any return type😃. Noticed? I will change that when you are done with some revisions😃😃. My bad. Apologies for that

I was just about to raise this issue. You can correct it before I proceed with the tech review.

**Comment by vinstex** [comment]
@BrianaNzivu, I fixed that. Thank you for your understanding.

**Comment by BrianaNzivu** [comment]
> @BrianaNzivu, I fixed that. Thank you for your understanding.

No worries.
Proceeding with tech review.

**Comment by BrianaNzivu** [comment]
Tech review done 👍

File: content/articles/android-autologout-kotlin/index.md [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+            android:id="@+id/sampleTxt"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintLeft_toLeftOf="parent"
+            app:layout_constraintRight_toRightOf="parent"
+            app:layout_constraintTop_toTopOf="parent" android:paddingLeft="15dp" android:paddingRight="15dp"/>
+
+</androidx.constraintlayout.widget.ConstraintLayout>
+```
+
+Run the app and do some touches, scrolls, and wait 15 seconds to check the feature. Also, do the same before the seconds elapse and check. Change the numbers and check too.
```
Comment (BrianaNzivu) [inline content/articles/android-autologout-kotlin/index.md:411]: Hey @vinstex .
Maybe you can link the repo that will contain this project's code here and a downloadable sample APK for this project linked which you can host on Google drive.

**Review by BrianaNzivu — APPROVED** [review]
Hey @vinstex   👋 .
Great job on the article! Interesting read and very minimal corrections to be made. Keep up the great job!
Just a final comment: 
- All images should be 600 x 400 px. Never forget that.
- Please go through all the changes I have made to avoid repeating the same mistakes in your future articles.

That's it! :smile: 

Wishing you all the best in your endeavors of becoming a peer reviewer. You will be a great addition to our team. If you make it in, please do not hesitate to ask for assistance in case you need some help because it is also a journey to becoming a great reviewer.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 

I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Kind Regards.

**Comment by vinstex** [comment]
Ta-da!

**Comment by vinstex** [comment]
@hectorkambow

**Comment by hectorkambow** [comment]
@section-engineering-education/content-moderator-team

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**